src/main/scala/com/databricks/spark/xml/XmlRelation.scala

```diff
@@ -67,6 +68,27 @@ case class XmlRelation protected[spark] (
       parseConf)
   }
 
+
+  override def buildScan(requiredColumns: Array[String]): RDD[Row] = {
+    val schemaFields = schema.fields
+    val requiredFields = StructType(requiredColumns.map(schema(_))).fields
+    if (schemaFields.deep == requiredFields.deep) {
+      buildScan
+    } else {
+      val safeRequiredFields = if (!parseConf.failFastFlag) {
+        // If `failFast` is disabled, then it needs to parse all the values
+        // so that we can drop malformed rows.
+        requiredFields ++ schemaFields.filterNot(requiredFields.contains(_))
+      } else {
+        requiredFields
+      }
+      StaxXmlParser.parse(
+        baseRDD(),
+        StructType(safeRequiredFields),
+        parseConf)
```

Comment:
When dropping rows with failing fast option disabled, I will return extra values as it needs to parse all the data to drop malformed rows. This will return correct results but I think I should slice the results before returning.